src/test/java/com/flightstats/hub/dao/aws/S3AccessMonitorTest.java

```diff
@@ -0,0 +1,75 @@
+package com.flightstats.hub.dao.aws;
+
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.DeleteVersionRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.when;
+
+public class S3AccessMonitorTest {
+    private final S3BucketName s3BucketName = new S3BucketName("test", "bucket");
+
+    @Test
+    public void testVerifyReadWriteAccess_errorPutObject_false() {
+        HubS3Client s3Client = mock(HubS3Client.class);
+        S3AccessMonitor monitor = new S3AccessMonitor(s3Client, s3BucketName);
+        doThrow(new RuntimeException("public void testVerifyReadWriteAccess_errorPutObject_false"))
+                .when(s3Client).putObject(any(PutObjectRequest.class));
+
+        assertFalse(monitor.verifyReadWriteAccess());
+    }
+
+    @Test
+    public void testVerifyReadWriteAccess_errorGetObject_false() {
+        HubS3Client s3Client = mock(HubS3Client.class);
+        S3AccessMonitor monitor = new S3AccessMonitor(s3Client, s3BucketName);
+        PutObjectResult putObjectResult = new PutObjectResult();
+        putObjectResult.setVersionId("testVersionId");
+        when(s3Client.putObject(any(PutObjectRequest.class))).thenReturn(putObjectResult);
+        doThrow(new RuntimeException("testVerifyReadWriteAccess_errorGetObject_false"))
+                .when(s3Client).getObject(any(GetObjectRequest.class));
+
+        assertFalse(monitor.verifyReadWriteAccess());
+    }
+
+
+    @Test
+    public void testVerifyReadWriteAccess_mockVersionId_true() {
+        HubS3Client s3Client = mock(HubS3Client.class);
+        S3AccessMonitor monitor = new S3AccessMonitor(s3Client, s3BucketName);
+        PutObjectResult putObjectResult = new PutObjectResult();
+        putObjectResult.setVersionId("testVersionId");
```

Comment:
This is repeated block of code in all tests. this can be moved to @Before junit method